util: Remove `fsbridge::get_filesystem_error_message()`

hebasto opened the pull request:
The `fsbridge::get_filesystem_error_message()` function exhibits several drawbacks:

1. It was introduced in https://github.com/bitcoin/bitcoin/pull/14192 to account for platform-specific variations in
`boost::filesystem::filesystem_error::what()`. Since [migrating](https://github.com/bitcoin/bitcoin/pull/20744) to `std::filesystem`, those discrepancies no longer exist.

2. It fails to display UTF-8 paths correctly on Windows:
```
> build\bin\Release\bitcoind.exe -datadir="C:\Users\hebasto\dd_₿_🏃" -regtest
...
2025-04-30T00:17:48Z DeleteAuthCookie: Unable to remove random auth cookie file: remove: Access is denied.: "C:\Users\hebasto\dd_?_??\regtest\.cookie"
...
```

3. It relies on `std::wstring_convert`, which was deprecated in C++17 and removed in C++26 (also see https://github.com/bitcoin/bitcoin/issues/32361).

This PR removes the obsolete `fsbridge::get_filesystem_error_message()` function, thereby resolving all of the above issues.

hebasto commented:
Friendly ping @ryanofsky @davidgumberg @hodlinator @sipsorcery :)

maflcko reviewed (approved):
lgtm ACK 671a9a258705b22b26cb8066fc0e53988c960fc6

The new error messages also read nicer. Left some further nits, while touching those lines.

hebasto committed:
util: Remove `fsbridge::get_filesystem_error_message()`

The `fsbridge::get_filesystem_error_message()` function exhibits several
drawbacks:

1. It was introduced in https://github.com/bitcoin/bitcoin/pull/14192 to
account for platform-specific variations in
`boost::filesystem::filesystem_error::what()`. Since migrating to
`std::filesystem`, those discrepancies no longer exist.

2. It fails to display UTF-8 paths correctly on Windows.

3. It relies on `std::wstring_convert`, which was deprecated in C++17
and removed in C++26.

This change removes the `fsbridge::get_filesystem_error_message()`
function, thereby resolving all of the above issues.

Additionally, filesystem error messages now use the "Warning" log level.

hebasto commented on src/rpc/request.cpp:
```diff
@@ -170,7 +170,7 @@ void DeleteAuthCookie()
         }
     } catch (const fs::filesystem_error& e) {
-        LogPrintf("%s: Unable to remove random auth cookie file: %s\n", __func__, fsbridge::get_filesystem_error_message(e));
+        LogPrintf("%s: Unable to remove random auth cookie file %s: %s\n", __func__, fs::PathToString(e.path1()), e.code().message());
```
Thanks! Updated.

hebasto commented:
The recent [feedback](https://github.com/bitcoin/bitcoin/pull/32383#discussion_r2068272988) from @maflcko has been addressed.

laanwj reviewed (approved):
Code review ACK 97eaadc3bf9f621ba397e29bb1c0cd99af55f2e3

This was sure some horrible code, i'm confused what it even does ~~(calling `MultiByteToWideChar` *twice* then do another pass of `std::codecvt_utf8_utf16` seems way overkill, it)~~ (edit: ok, it does makes sense, it's still awful though). Glad to replace it with a standardized method.

davidgumberg commented on src/rpc/request.cpp:
```diff
@@ -170,7 +170,7 @@ void DeleteAuthCookie()
         }
     } catch (const fs::filesystem_error& e) {
-        LogPrintf("%s: Unable to remove random auth cookie file: %s\n", __func__, fsbridge::get_filesystem_error_message(e));
+        LogWarning("Unable to remove random auth cookie file %s: %s\n", fs::PathToString(e.path1()), e.code().message());
```
non blocking nit, I think we can use `e.what()` in place of `e.code().message()`

hebasto replied:
UTF-8 paths included in `e.what()` are not displayed correctly on Windows.

laanwj replied:
> UTF-8 paths included in e.what() are not displayed correctly on Windows.

Maybe they will after #32380? In any case, including the filename doesn't seem necessary.

davidgumberg commented:
untested crACK https://github.com/bitcoin/bitcoin/pull/32383/commits/97eaadc3bf9f621ba397e29bb1c0cd99af55f2e3

This code made sense when using boost, but I think `std::filesystem` implementations are responsible for and better equipped to handle platform specific character width issues.

achow101 commented:
ACK 97eaadc3bf9f621ba397e29bb1c0cd99af55f2e3
